Detectors/TPC/workflow/src/CATrackerSpec.cxx

```diff
@@ -73,9 +86,132 @@ DataProcessorSpec getCATrackerSpec(bool processMC, std::vector<int> const& input
       auto& parser = processAttributes->parser;
       auto& tracker = processAttributes->tracker;
       parser = std::make_unique<ClusterGroupParser>();
-      tracker = std::make_unique<o2::tpc::TPCCATracking>();
-      if (tracker->initialize(options.c_str()) != 0) {
-        throw std::invalid_argument("TPCCATracking initialization failed");
+      tracker = std::make_unique<o2::tpc::GPUCATracking>();
+
+      // Prepare initialization of CATracker - we parse the deprecated option string here,
+      // and create the proper configuration objects for compatibility.
+      // This should go away eventually.
+
+      // Default Settings
+      float solenoidBz = -5.00668; // B-field
+      float refX = 83.;            // transport tracks to this x after tracking, >500 for disabling
+      bool continuous = false;     // time frame data v.s. triggered events
+      int nThreads = 1;            // number of threads if we run on the CPU, 1 = default, 0 = auto-detect
+      bool useGPU = false;         // use a GPU for processing, if false uses GPU
+      bool dump = false;           // create memory dump of processed events for standalone runs
+      char gpuType[1024] = "CUDA"; // Type of GPU device, if useGPU is set to true
+      GPUDisplayBackend* display = nullptr;
+
+      const auto grp = o2::parameters::GRPObject::loadFrom("o2sim_grp.root");
+      if (grp) {
+        LOG(INFO) << "Initializing run paramerers from GRP";
+        solenoidBz *= grp->getL3Current() / 30000.;
+        continuous = grp->isDetContinuousReadOut(o2::detectors::DetID::TPC);
+      } else {
+        LOG(ERROR) << "Failed to initialize run parameters from GRP";
+        // should we call fatal here?
+      }
+
+      // Parse the config string
+      const char* opt = options.c_str();
```

Comment:
Agreed, but as I said, this was just a temporary solution to clean up the interface. I'd do away with this string parsing here completely, and use proper command line options of the workflow. How this is organized, I don't really care, but I wouldn't necessarily put it inside `GPUO2InterfaceConfiguration`. But I think this can be cleaned up later. Meaning of this PR was to define and clean up the interface between the spec and GPUCAReconstruction.